### DISCUSSION TITLE
[Enhancement] New window with NavigationPage support

### DISCUSSION THREAD

**Discussion by JohnH-FF** [dotnet/maui#53] [General]
## Summary
XF doesn't natively support multiple windows, make it so number 1!
Currently have to drop into platform specific code (via DI) to implement this, and provides a sub optimal solution.

## API Changes

INavigation.PushToNewWindowAsync(Page page);

## Intended Use Case
On desktop platforms this will push the page (or NavigationPage) into a new window. On mobile it will perform the same as INavigation.PushAsync until such time that mobile supports multi window (if ever).

**Comment by freever** [comment]
On mobile this could push to a modal dialog...?

**Comment by JohnH-FF** [comment]
@freever I was thinking that before I posted, but the better minds than mine at MS could decide if that workflow works. :)

**Comment by GalaxiaGuy** [comment]
Perhaps there should be an options parameter that could include `PreferNewWindow` leaving it up to the platform to handle? It could even replace the `bool animated` parameter:

```csharp
[Flags]
public enum NavigationOptions
{
    None = 0,
    Animated = 1,
    PreferNewWindow = 2,
}
```
